fpm/src/fpm.f90

```diff
@@ -14,16 +13,134 @@ module fpm
                        resolve_module_dependencies
 use fpm_manifest, only : get_package_data, default_executable, &
     default_library, package_t, default_test
-use fpm_error, only : error_t
+use fpm_error, only : error_t, fatal_error
 use fpm_manifest_test, only : test_t
-use,intrinsic :: iso_fortran_env, only : stderr=>error_unit
+use,intrinsic :: iso_fortran_env, only : stdin=>input_unit,   &
+                                       & stdout=>output_unit, &
+                                       & stderr=>error_unit
+use fpm_manifest_dependency, only: dependency_t
 implicit none
 private
 public :: cmd_build, cmd_install, cmd_run, cmd_test
 
 contains
 
 
+recursive subroutine add_libsources_from_package(sources,package_list,package, &
+                                                  package_root,dev_depends,error)
+    ! Discover library sources in a package, recursively including dependencies
+    !
+    type(srcfile_t), allocatable, intent(inout), target :: sources(:)
+    type(string_t), allocatable, intent(inout) :: package_list(:)
+    type(package_t), intent(in) :: package
+    character(*), intent(in) :: package_root
+    logical, intent(in) :: dev_depends
+    type(error_t), allocatable, intent(out) :: error
+
+    ! Add package library sources
+    if (allocated(package%library)) then
+
+        call add_sources_from_dir(sources, join_path(package_root,package%library%source_dir), &
+                                    FPM_SCOPE_LIB, error=error)
+
+        if (allocated(error)) then
+            return
+        end if
+
+    end if
+
+    ! Add library sources from dependencies
+    if (allocated(package%dependency)) then
+
+        call add_dependencies(package%dependency)
+
+        if (allocated(error)) then
+            return
+        end if
+
+    end if
+
+    ! Add library sources from dev-dependencies
+    if (dev_depends .and. allocated(package%dev_dependency)) then
```

Comment:
Is this fetching the `[dev-dependencies]` for a package you depend on? Because the idea is that those are dependencies NOT needed for building the library, so we shouldn't fetch and build those recursively.

Reply:
Nevermind, I see it now. You're using the logical input `dev_depends` to take care of it.